### DISCUSSION TITLE
How to use PnP.Core Authentication without using Azure AD for SharePoint On-Premise

### DISCUSSION THREAD

**Discussion by dmta2005** [pnp/pnpcore#1163] [Q&A] [answered]
I would like to know how to upload a Word document to a SharePoint 2019 On-Premise from an ASP.NET Core 6 Web Application using PnP.Core.   I’m not going to register my application in Azure Active Directory.  

How do I create a PnpContext to connect to the SharePoint site so I can insert the record? if yes, please let me know where I can find the documentation how to do it.

Can I connect to SharePoint Site SharePoint Client Id & Client Secret to authenticate using PnP.Core? if yes, please let me know where I can find the documentation how to do it.

I would really appreciate any suggestions.
Thanks

**Comment by jansenbe** [comment, marked as answer]
@dmta2005 : PnP Core SDK does not support on-premises SharePoint, only SharePoint Online is supported. For developing against on-premises you can either manually use the REST API or CSOM APIs (https://www.nuget.org/packages/Microsoft.SharePoint2019.CSOM).

**Reply by dmta2005** [reply]
Thank you so much for replying so quickly.  I was able to use the Web.API .NET 4.8 to access the SP 2019 data.
